jscomp/others/jsxDOM.res

```diff
@@ -0,0 +1,621 @@
+/* Copyright (C) 2015-2016 Bloomberg Finance L.P.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * In addition to the permissions granted to you by the LGPL, you may combine
+ * or link a "work that uses the Library" with a publicly distributed version
+ * of this file to produce a combined library or application, then distribute
+ * that combined work under the terms of your choosing, with no requirement
+ * to comply with the obligations normally placed on you by section 4 of the
+ * LGPL version 3 (or the corresponding section of a later version of the LGPL
+ * should you choose to use a later version).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. */
+
+type style = JsxDOMStyle.t
+type domRef
+
+/*
+  This list isn't exhaustive. We'll add more as we go.
+*/
+type domProps = {
+  key?: string,
+  ref?: domRef,
+  /* accessibility */
+  /* https://www.w3.org/TR/wai-aria-1.1/ */
+  /* https://accessibilityresources.org/<aria-tag> is a great resource for these */
+  @as("aria-current")
+  ariaCurrent?: [#page | #step | #location | #date | #time | #"true" | #"false"],
+  @as("aria-details")
+  ariaDetails?: string,
+  @as("aria-disabled")
+  ariaDisabled?: bool,
+  @as("aria-hidden")
+  ariaHidden?: bool,
+  @as("aria-invalid") ariaInvalid?: [#grammar | #"false" | #spelling | #"true"],
+  @as("aria-keyshortcuts")
+  ariaKeyshortcuts?: string,
+  @as("aria-label")
+  ariaLabel?: string,
+  @as("aria-roledescription")
+  ariaRoledescription?: string,
+  /* Widget Attributes */
+  @as("aria-autocomplete") ariaAutocomplete?: [#inline | #list | #both | #none],
+  /* https://www.w3.org/TR/wai-aria-1.1/#valuetype_tristate */
+  @as("aria-checked")
+  ariaChecked?: [
```

Comment:
Unified formatting across this file would be nice. I.e. have this on one line, and run the formatter over the file.

Reply:
Oh I ran the formatter too, in my case it added a line-break 🤷‍♂️

Reply:
I'll make it to one line, it looks better.